src/cluster_legacy.c

```diff
@@ -6543,7 +6556,30 @@ unsigned int delKeysInSlot(unsigned int hashslot, int lazy, bool propagate_del,
      * state so that we don't assert in propagateNow(). */
     server.server_del_keys_in_slot = 1;
     unsigned int j = 0;
+    robj *argv[4];
     int before_execution_nesting = server.execution_nesting;
+    enterExecutionUnit(1, 0);
+
+    /* Propagate as a single CLUSTER FLUSHSLOT <slot> ASYNC/SYNC. */
+    if (propagate_del) {
```

Comment:
Do the hooks fire if the replica executes flushslot? Couldn't the module hooks also need to get executed on the replica? I would imagine something like search might break.

Reply:
The replica might also not support `CLUSTER FLUSHSLOT`, we should check the version of the replica. Otherwise it's a backwards incompatible change.

Reply:
> I guess there is a discrepancy where the primary would not fire keyspace events to clients, but the replica would. So we should align those behaviors.

Yeah. I agree that consistency is more important.

Reply:
I guess this is the existing behavior, since delKeysInSlot will not generate keyspace events on the primary, but the replicated `UNLINK` will. Triggering `CLUSTER FLUSHSLOT` on the replica is the same effect.